src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -51,83 +125,161 @@ public class ScriptTagHelper : TagHelper
-                }
-                else
+                builder.Append("document.write(\"");
+
+                foreach (var src in fallbackSrcs)
                 {
-                    AppendSrc(content, attribute.Key);
+                    builder.Append("<script");
+
+                    if (!attributes.ContainsKey("src"))
+                    {
+                        AppendSrc(builder, "src", src);
+                    }
+
+                    foreach (var attribute in attributes)
+                    {
+                        if (!attribute.Key.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var encodedKey = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Key);
+                            var encodedValue = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Value);
+
+                            builder.AppendFormat(CultureInfo.InvariantCulture, " {0}=\\\"{1}\\\"", encodedKey, encodedValue);
+                        }
+                        else
+                        {
+                            AppendSrc(builder, attribute.Key, src);
+                        }
+                    }
+
+                    builder.Append("><\\/script>");
                 }
+                
+                builder.Append("\"));</script>");
             }
+        }
 
-            content.Append("><\\/script>\"));</script>");
+        private void EnsureGlobbingUrlBuilder()
+        {
+            if (GlobbingUrlBuilder == null)
+            {
+                GlobbingUrlBuilder = new GlobbingUrlBuilder(
+                    HostingEnvironment.WebRootFileProvider,
+                    Cache,
+                    ViewContext.HttpContext.Request.PathBase);
+            }
+        }
+
+        private static void BuildScriptTag(IDictionary<string, string> attributes, string content, StringBuilder builder)
+        {
+            builder.Append("<script");
+
+            foreach (var attribute in attributes)
+            {
+                builder.AppendFormat(CultureInfo.InvariantCulture, " {0}=\"{1}\"", attribute.Key, attribute.Value);
+            }
 
-            output.Content = content.ToString();
+            builder.Append(">");
+            builder.Append(content);
+            builder.Append("</script>");
         }
 
-        private void AppendSrc(StringBuilder content, string srcKey)
+        private void AppendSrc(StringBuilder content, string srcKey, string srcValue)
         {
             // Append src attribute in the original place and replace the content the fallback content
             // No need to encode the key because we know it is "src".
             content.Append(" ")
                    .Append(srcKey)
                    .Append("=\\\"")
-                   .Append(WebUtility.HtmlEncode(FallbackSrc))
+                   .Append(WebUtility.HtmlEncode(srcValue))
```

Comment:
Yeah I specifically didn't change it due to there being a separate issue to clean them up, and again because I didn't want to conflate what this PR was about (copying from `LinkTagHelper`).

I can't activate an `IHtmlEncoder` property until that PR goes in, as it registers them in the container via `MvcServices`. I can sync up @ajaybhargavb if my PR goes in first so that he updates `ScriptTagHelper` as well.